docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+
+Service catalog controllers vary, however, from ingress controllers, because a cluster can consist of many different controllers, each of which handles actions on a subset of the available service-related resources. For example, a cluster can run a service controller to handle AWS RDS databases, and a completely separate one to handle AWS Elastic Load Balancer instances.
+
+As we'll detail below, each service controller implementation must to conform to an implementation spec, but that spec allows for a heterogenous set of controllers in a cluster. We believe this allowance encourages flexibility in a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how a `StorageClass` defines a _kind_ of persistent volume, but we expect that, in practice, the kinds of services represented here will be more coarse grained than the kinds of volumes represented in `StorageClass`es
+  - References a service ID and plan ID, which is managed and understood by a service controller (see below)
+  - Created by an operator or tool
+- `ServiceInstance`
+  - An instance of a `ServiceClass` -- an in-cluster representation of a provisioned service
+  - Can be statically provisioned by an operator, to enable multiple consumers to bind to it (enables multi-tenancy)
+  - Can be dynamically provisioned and bound to by a consumer, to guarantee an app’s exclusive use of the resource (guarantees single-tenancy)
+- `ServiceInstanceClaim`
+  - Created by a consumer to bind to an existing ServiceInstance by name. As written above, this workflow enables multi-tenant usage of a resource
+  - Created by a consumer to provision and bind to an existing ServiceClass. As written above, this workflow enables dedicated, single-tenant usage of a resource
+- `ServiceInstanceBinding`
+  - Created in response to a successful bind operation
+  - References credentials created as a result of the bind operation
+  - Holds a record of an application that’s bound to a service
+
+
+The below diagram shows the interactions between these resources:
+
+![Resource Interactions](./img/partial-flow.png)
+
+# Example Resources
+
+## ServiceClass
+
+```yaml
+kind: ServiceClass
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: postgres-small
+spec:
+  serviceID: c5a0ad8e-b57b…
+  planID: c68cd6b1-be9f…
+```
+
+## ServiceInstance
+
+```yaml
+kind: ServiceInstance
+apiVersion: service-catalog.k8s.io/alpha
+metadata:
+  name: zip-code-db
+spec:
+  serviceClass: postgres-small
+```
+
```

Comment:
I think we definitely need a way to make a service in one namespace visible to others.  (the service entry itself, not just the ability to connect to the IP).  If we make sure that works, do we get past this "multi-tenancy vs. single tenancy as part of the instantiation" need?